### ISSUE TITLE
Incorrect report of: expected &&{unknown}, found &Q

### ISSUE THREAD

**Issue by yunyi1201** [rust-lang/rust-analyzer#19227]
The attached code incorrectly reports a unexpected type.
This is only shown in-IDE (VScode). It is not shown with cargo check or cargo clippy. (There is no visible issue at runtime.)
In-IDE error:
> expected &&{unknown}, found &Q. rust-analyzer[E0308](https://doc.rust-lang.org/stable/error_codes/E0308.html)

**rust-analyzer version**:  0.3.2319-standalone (b3ee3a96a2 2025-02-24)

**rustc version**: 1.87.0-nightly (f280acf4c 2025-02-19)

**code snippet to reproduce**: 
```rust
use std::borrow::Borrow;

struct Key<T> {
    data: T,
}

impl<T> Key<T>
where
    T: Ord,
{
    fn borrow<Q>(&self, k: &Q)
    where
        T: Borrow<Q>,
        Q: Ord,
    {
        let _ = self.data.borrow().cmp(k);
    }
}
```

**Comment by ChayimFriedman2** [comment]
This is a known problem.
